peps/pep-0807.rst

```diff
@@ -0,0 +1,398 @@
+PEP: 807
+Title: Index support for Trusted Publishing
+Author: William Woodruff <william@yossarian.net>
+Sponsor: Donald Stufft <donald@stufft.io>
+PEP-Delegate: Donald Stufft <donald@stufft.io>
+Status: Draft
+Type: Standards Track
+Topic: Packaging
+Created: 15-Sep-2025
+Post-History: `08-Aug-2025 <https://discuss.python.org/t/pre-pep-trusted-publishing-token-exchange/103067>`__
+
+Abstract
+========
+
+This PEP proposes a standard mechanism through which arbitrary
+Python package indices can support "Trusted Publishing," a misuse-resistant
+credential exchange scheme already implemented by the Python Package Index
+(PyPI).
+
+The mechanism proposed in this PEP is designed to encapsulate PyPI's
+`existing implementation <https://docs.pypi.org/trusted-publishers/>`_
+of Trusted Publishing, while allowing other indices to implement the same
+scheme in a manner that is discoverable by and interoperable with existing
+Python package uploading clients.
+
+Rationale and Motivation
+========================
+
+"Trusted Publishing" is PyPI's term of art for using the
+`OpenID Connect (OIDC) standard <https://openid.net/connect/>`_
+to exchange a short-lived *identity credential* from a trusted
+third-party service (like a CI/CD or cloud provider) for a short-lived,
+minimally-scoped *upload credential* that can be used to publish
+to the index.
+
+Trusted Publishing was originally designed and enabled on PyPI in 2023 as
+a non-standard (PyPI-specific) feature, much like the existing
+`upload API <https://docs.pypi.org/trusted-publishers/>`_. It has seen
+widespread adoption in that capacity: over 640,000 files have been published
+to PyPI using a Trusted Publisher (as of September 2025), and PyPI's
```

Comment:
Might be useful to add proportion of total, 640k is a big number but this could be 5% of all uploads or 95%!